File: Dsa-day164.java
class Solution {
    public int countNegatives(int[][] grid) {
        if (grid.length == 0 || grid[0].length == 0) return 0;
        int m = grid.length, n = grid[0].length;
        int count = 0;
        for (int[] row : grid) {
            int left = 0, right = n;
            while (left < right) {
                int mid = left + (right - left) / 2;
                if (row[mid] < 0) {
                    right = mid;
                } else {
                    left = mid + 1;
                }
            }
            count += n - left;
        }
        return count;
    }
}

File: Dsa-day164.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        if not grid or not grid[0]:
            return 0
        m, n = len(grid), len(grid[0])
        count = 0
        for row in grid:
            left, right = 0, n
            while left < right:
                mid = (left + right) // 2
                if row[mid] < 0:
                    right = mid
                else:
                    left = mid + 1
            count += n - left
        return count